SofaKernel/modules/SofaBaseTopology/PointSetTopologyContainer.cpp

```diff
@@ -33,6 +36,25 @@ namespace component
 namespace topology
 {
 
+namespace
+{
+
+struct GeneratePointID
+{
+    typedef sofa::core::topology::BaseMeshTopology::PointID PointID;
+
+    GeneratePointID( PointID startId = PointID(0) )
+    :current(startId)
+    {
+    }
+
+    PointID operator() () { return current++; }
+
+    PointID current;
+};
```

Comment:
Just for curiosity, why using this structure instead of a simple int counter ? Is it for the algorithm to allocate consecutive memory?